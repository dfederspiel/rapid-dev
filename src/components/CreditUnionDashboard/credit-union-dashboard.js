import React from 'react';
import { Row, Column } from "simple-flexbox";

export default class CreditUnionDashboard extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="credit-union-dashboard">
                <Row justifyContent="space-between">
                    <Column>
                        <label>Pending Transactions</label>
                        <a href="">{this.props.pendingTransactions}</a>
                    </Column>
                    <div className="vl"/>
                    <Column>
                        <label>Pending Acknowledgements</label>
                        <a href="">{this.props.pendingAcknowledgements}</a>
                    </Column>
                    <div className="vl" />
                    <Column>
                        <label>Pending Templates</label>
                        <a href="">{this.props.pendingTemplates}</a>
                    </Column>
                </Row>
            </div>
        );
    }
}

