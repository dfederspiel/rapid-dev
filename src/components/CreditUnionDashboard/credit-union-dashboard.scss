@import '../../styles/variables';
@import '../../styles/mixins';

.credit-union-dashboard {
    padding: 2rem;

    label {
        font-weight: $font-weight-medium;
        margin-bottom: 2rem;
    }

    a {
        font-size: 3.3rem;
        font-weight: $font-weight-bold;
    }

    .vl {
        border-left: .2rem solid $light-background;
        height: 10rem;
    }
}
