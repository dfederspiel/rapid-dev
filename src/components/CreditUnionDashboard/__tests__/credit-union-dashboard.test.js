import React from 'react';
import CreditUnionDashboard from '../credit-union-dashboard';
import renderer from 'react-test-renderer';

const creditUnionDashboardProps = {
    pendingTransactions: 607,
    pendingTemplates: 45,
    pendingAcknowledgements: 13
};

describe('the credit union component', () => {
    
    it('renders correctly', () => {
        const tree = renderer.create(
            <CreditUnionDashboard
                {...creditUnionDashboardProps}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});