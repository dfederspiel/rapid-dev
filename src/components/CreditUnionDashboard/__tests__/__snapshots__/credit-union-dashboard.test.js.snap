// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the credit union component renders correctly 1`] = `
<div
  className="credit-union-dashboard"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label>
        Pending Transactions
      </label>
      <a
        href=""
      >
        607
      </a>
    </div>
    <div
      className="vl"
    />
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label>
        Pending Acknowledgements
      </label>
      <a
        href=""
      >
        13
      </a>
    </div>
    <div
      className="vl"
    />
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
        }
      }
    >
      <label>
        Pending Templates
      </label>
      <a
        href=""
      >
        45
      </a>
    </div>
  </div>
</div>
`;
